### PR TITLE
Fixes a bug in the S3 sink where events without handles throw NPE

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Fixes a bug in the `s3` sink for null Event handles.

```
2023-06-02T22:22:12,052 [performance-alb-pipeline-sink-worker-2-thread-3] ERROR org.opensearch.dataprepper.plugins.sink.S3SinkService - Exception while write event into buffer :
java.lang.NullPointerException: Cannot invoke "org.opensearch.dataprepper.model.event.EventHandle.release(boolean)" because "eventHandle" is null
	at org.opensearch.dataprepper.plugins.sink.S3SinkService.releaseEventHandles(S3SinkService.java:139) ~[s3-sink-2.3.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.plugins.sink.S3SinkService.output(S3SinkService.java:120) ~[s3-sink-2.3.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.plugins.sink.S3Sink.doOutput(S3Sink.java:106) ~[s3-sink-2.3.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.model.sink.AbstractSink.lambda$output$0(AbstractSink.java:64) ~[data-prepper-api-2.3.0-SNAPSHOT.jar:?]
	at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:141) ~[micrometer-core-1.10.5.jar:1.10.5]
	at org.opensearch.dataprepper.model.sink.AbstractSink.output(AbstractSink.java:64) ~[data-prepper-api-2.3.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.pipeline.Pipeline.lambda$publishToSinks$5(Pipeline.java:336) ~[data-prepper-core-2.3.0-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
```
 
### Issues Resolved
N/A
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3SinkService.java [hunk 1]
```diff
@@ -106,7 +106,9 @@ void output(Collection<Record<Event>> records) {
                 final byte[] encodedBytes = encodedEvent.getBytes();
 
                 currentBuffer.writeEvent(encodedBytes);
-                bufferedEventHandles.add(event.getEventHandle());
```
Comment (cmanning09) [inline data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3SinkService.java:109]: Should we add a unit test to confirm this is fixed? And help prevent future NPEs around this leak into this plugin?

Comment (dlvenable) [reply data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3SinkService.java:109]: I actually tried this, but I was not able to get it to crash such that the test would actually show any verification.

Comment (kkondaka) [reply data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3SinkService.java:109]: Null event handle would be the default case when not using end-to-end acknowledgements.

**Review by kkondaka — APPROVED** [review]

File: data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3SinkService.java (hunk 1 above)
Comment (dlvenable) [reply data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3SinkService.java:109]: @cmanning09 , I found that there was an unnecessary try-catch on `NullPointerException`. I removed that and was able to add a valid test.

**Review by chenqi0805 — APPROVED** [review]

**Pull request merged**